File: lib/base/net/domain/domain_config.dart

import 'package:json_annotation/json_annotation.dart';

part 'domain_config.g.dart';

@JsonSerializable()
class DomainConfig {
  @JsonKey(name: 'lastDomain')
  String? lastDomain;

  @JsonKey(name: 'userAgent')
  String? userAgent;

  @JsonKey(name: 'adImageUrl')
  String? adImageUrl;

  @JsonKey(name: 'adClickUrl')
  String? adClickUrl;

  @JsonKey(name: 'adCountdown')
  int? adCountdown;

  @JsonKey(name: 'abUrl')
  String? abUrl;

  @JsonKey(name: 'domainList')
  List<String>? domainList;

  @JsonKey(name: 'dnsApis')
  List<String>? dnsApis;

  @JsonKey(name: 'configApis')
  List<String>? configApis;

  DomainConfig(
    this.lastDomain,
    this.adImageUrl,
    this.adClickUrl,
    this.adCountdown,
    this.abUrl,
    this.userAgent,
    this.domainList,
    this.dnsApis,
    this.configApis,
  );

  factory DomainConfig.fromJson(Map<String, dynamic> srcJson) =>
      _$DomainConfigFromJson(srcJson);

  Map<String, dynamic> toJson() => _$DomainConfigToJson(this);
}

File: android/ui/src/main/java/com/ui/DomainConfig.java
package com.ui;

import java.util.List;


public class DomainConfig {
    private String lastDomain;
    private String adImageUrl;
    private String adClickUrl;
    private Integer adCountdown;
    private String abUrl;
    private String userAgent;
    private List<String> domainList;
    private List<String> dnsApis;
    private List<String> configApis;
    private List<String> whiteList;


    public DomainConfig(String lastDomain, String adImageUrl, String adClickUrl, Integer adCountdown, String abUrl, String userAgent, List<String> domainList, List<String> dnsApis, List<String> configApis,List<String> whiteList) {
        this.lastDomain = lastDomain;
        this.adImageUrl = adImageUrl;
        this.adClickUrl = adClickUrl;
        this.adCountdown = adCountdown;
        this.abUrl = abUrl;
        this.userAgent = userAgent;
        this.domainList = domainList;
        this.dnsApis = dnsApis;
        this.configApis = configApis;
        this.whiteList = whiteList;
    }

    public String getLastDomain() {
        return lastDomain;
    }

    public void setLastDomain(String lastDomain) {
        this.lastDomain = lastDomain;
    }

    public String getAdImageUrl() {
        return adImageUrl;
    }

    public void setAdImageUrl(String adImageUrl) {
        this.adImageUrl = adImageUrl;
    }

    public String getAdClickUrl() {
        return adClickUrl;
    }

    public void setAdClickUrl(String adClickUrl) {
        this.adClickUrl = adClickUrl;
    }

    public Integer getAdCountdown() {
        return adCountdown;
    }

    public void setAdCountdown(Integer adCountdown) {
        this.adCountdown = adCountdown;
    }

    public String getAbUrl() {
        return abUrl;
    }

    public void setAbUrl(String abUrl) {
        this.abUrl = abUrl;
    }

    public String getUserAgent() {
        return userAgent;
    }

    public void setUserAgent(String userAgent) {
        this.userAgent = userAgent;
    }

    public List<String> getDomainList() {
        return domainList;
    }

    public void setDomainList(List<String> domainList) {
        this.domainList = domainList;
    }

    public List<String> getDnsApis() {
        return dnsApis;
    }

    public void setDnsApis(List<String> dnsApis) {
        this.dnsApis = dnsApis;
    }

    public List<String> getConfigApis() {
        return configApis;
    }

    public void setConfigApis(List<String> configApis) {
        this.configApis = configApis;
    }

    public List<String> getWhiteList() {
        return whiteList;
    }

    public void setWhiteList(List<String> whiteList) {
        this.whiteList = whiteList;
    }
}
